watertap/property_models/unit_specific/anaerobic_digestion/adm1_reactions.py

```diff
@@ -1459,7 +1461,7 @@ def Dissociation_rule(self, t):
 
         # Equation from [2]
         def CO2_acid_base_equilibrium_rule(self, t):
-            return pyo.log(10**-self.pK_a_co2) == (
+            return pyo.log(10**-self.pK_a_co2 + eps) == (
```

Comment:
This constraint looks really odd to me - why not do `-pK_a_CO2 == -6.35*exp(7646/R*(1/Tref-T))` (or something to that effect)? At the moment you are taking the log of an exponent which seems to be redundant.

Also, is the `eps` really needed? If you are trying to avoid evaluation errors then bounding the terms in the logs and divisions would generally be sufficient.

Reply:
We could simplify it to `log(10) * -pK_a_CO2 == ...` in which case the `eps `probably wouldn't be needed. The LHS wouldn't just be `pK_a_co2` because `log` is actually the natural log, right?

`pK_a_co2` currently is bounded in the domain of `PositiveReals`, but it was still getting a potential evaluation error